src/main/java/wolox/training/models/Book.java

```diff
@@ -86,70 +87,72 @@ public String getAuthor() {
     }
 
     public void setAuthor(String author) {
-        this.author = author;
+        this.author = Preconditions.checkNotNull(author, "The author can't be null");
```

Comment:
Change all preconditions in Strings to check if string is null OR empty.
For that, use checkArgument(var != null and var.!isEmpty())
You wont be able to assign the return of that, so first line checks, next line assigns.